feature/platform/transceiver/otg_test/zr_temperature_test/zr_temperature_test.go

```diff
@@ -0,0 +1,145 @@
+// Copyright 2022 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package zr_temperature_test
+
+import (
+	"reflect"
+	"testing"
+	"time"
+
+	"github.com/openconfig/featureprofiles/internal/fptest"
+	"github.com/openconfig/ondatra"
+	"github.com/openconfig/ondatra/gnmi"
+	"github.com/openconfig/ondatra/gnmi/oc"
+	"github.com/openconfig/ygnmi/ygnmi"
+	"github.com/openconfig/ygot/ygot"
+)
+
+func TestMain(m *testing.M) {
+	fptest.RunTests(m)
+}
+
+// Topology:
+//   dut1:port1 <--> port1:dut2
+//
+
+func TestZRTemperatureState(t *testing.T) {
+	dut1 := ondatra.DUT(t, "dut1")
+	dp1 := dut1.Port(t, "port1")
+	t.Logf("dut1: %v", dut1)
+	t.Logf("dut1 dp1 name: %v", dp1.Name())
+	// Check if TRANSCEIVER is of type 400ZR
+	if dp1.PMD() != ondatra.PMD400GBASEZR {
+		t.Fatalf("%s Transceiver is not 400ZR its of type: %v", dp1, dp1.PMD())
+	}
+	component1 := gnmi.OC().Component(dp1.Name())
+	temperatureInstantdp1 := gnmi.Get(t, dut1, component1.Temperature().Instant().State())
+	// Check temperature return value of correct type
+	if reflect.TypeOf(temperatureInstantdp1).Kind() != reflect.Float64 {
+		t.Fatalf("Return value is not type float64")
+	}
+	t.Logf("Port1 dut1 %s Instant Temperature: %v", dp1.Name(), temperatureInstantdp1)
+	temperatureMin := gnmi.Get(t, dut1, component1.Temperature().Min().State())
+	t.Logf("Port1 dut1 %s Min Temperature: %v", dp1.Name(), temperatureMin)
+	if reflect.TypeOf(temperatureMin).Kind() != reflect.Float64 {
+		t.Fatalf("Return value is not type float64")
+	}
+	temperatureMax := gnmi.Get(t, dut1, component1.Temperature().Max().State())
+	t.Logf("Port1 dut1 %s Max Temperature: %v", dp1.Name(), temperatureMax)
+	if reflect.TypeOf(temperatureMax).Kind() != reflect.Float64 {
+		t.Fatalf("Return value is not type float64")
+	}
+	temperatureAvg := gnmi.Get(t, dut1, component1.Temperature().Avg().State())
+	if reflect.TypeOf(temperatureAvg).Kind() != reflect.Float64 {
+		t.Fatalf("Return value is not type float64")
+	}
+	t.Logf("Port1 dut1 %s Avg Temperature: %v", dp1.Name(), temperatureAvg)
+	// temperature average should be between min and max
```

Comment:
what do you think about adding a quick check that the values look reasonable. These are in Celsius, so perhaps they should all be above 0 and below 300 degrees as extremes?

Reply:
Hi Eric,

Added check what you suggested kindly review.

Thanks

Reply:
Thanks.